java/src/main/java/com/google/oak/crypto/ClientEncryptor.java

```diff
@@ -74,39 +93,65 @@ public ClientEncryptor(final byte[] serializedServerPublicKey) {
+    }
+    EncryptedRequest encryptedRequest = encryptedRequestBuilder.build();
+
     // TODO(#3843): Return unserialized proto messages once we have Java encryption without JNI.
-    // TODO(#3642): Implement Java Hybrid Encryption.
-    EncryptedRequest encryptedRequest =
-        EncryptedRequest.newBuilder()
-            .setEncryptedMessage(AeadEncryptedMessage.newBuilder()
-                                     .setCiphertext(ByteString.copyFrom(plaintext))
-                                     .setAssociatedData(ByteString.copyFrom(associatedData))
-                                     .build())
-            .setSerializedEncapsulatedPublicKey(ByteString.copyFrom(TEST_ENCAPSULATED_PUBLIC_KEY))
-            .build();
     return Result.success(encryptedRequest.toByteArray());
   }
 
   /**
-   * Decrypts a [`EncryptedResponse`] proto message using AEAD.
+   * Decrypts a {@code EncryptedResponse} proto message using AEAD.
    * <https://datatracker.ietf.org/doc/html/rfc5116>
    *
-   * @param encryptedResponse a serialized {@code EncryptedResponse} message
+   * @param serializedEncryptedResponse a serialized {@code EncryptedResponse} message
    * @return a response message plaintext and associated data wrapped in a {@code Result}
    */
   @Override
   public final Result<Encryptor.DecryptionResult, Exception> decrypt(
-      final byte[] encryptedResponse) {
-    // TODO(#3843): Accept unserialized proto messages once we have Java encryption without JNI.
-    // TODO(#3642): Implement Java Hybrid Encryption.
+      final byte[] serializedEncryptedResponse) {
+    // Deserialize response message.
+    EncryptedResponse encryptedResponse;
     try {
-      AeadEncryptedMessage aeadEncryptedMessage =
-          EncryptedResponse.parseFrom(encryptedResponse).getEncryptedMessage();
-      byte[] plaintext = aeadEncryptedMessage.getCiphertext().toByteArray();
-      byte[] associatedData = aeadEncryptedMessage.getAssociatedData().toByteArray();
-      return Result.success(new ClientEncryptor.DecryptionResult(plaintext, associatedData));
+      encryptedResponse = EncryptedResponse.parseFrom(serializedEncryptedResponse);
     } catch (InvalidProtocolBufferException e) {
       return Result.error(e);
     }
+    AeadEncryptedMessage aeadEncryptedMessage = encryptedResponse.getEncryptedMessage();
+    byte[] ciphertext = aeadEncryptedMessage.getCiphertext().toByteArray();
+    byte[] associatedData = aeadEncryptedMessage.getAssociatedData().toByteArray();
+
+    // Decrypt response.
+    Result<byte[], Exception> openResult =
+        this.senderResponseContext.open(ciphertext, associatedData);
+    if (openResult.isError()) {
+      return Result.error(openResult.error().get());
+    }
+    byte[] plaintext = openResult.success().get();
+
+    // TODO(#3843): Accept unserialized proto messages once we have Java encryption without JNI.
+    return Result.success(new ClientEncryptor.DecryptionResult(plaintext, associatedData));
```

Comment:
All this can also be simplified to one line.

Reply:
`ClientEncryptor.DecryptionResult` accepts 2 arguments and `this.senderResponseContext.open` returns only 1